storage/innobase/trx/trx0sys.cc

```diff
@@ -577,8 +577,6 @@ void trx_sys_create_sys_pages(void) {
 /*********************************************************************
 Shutdown/Close the transaction system. */
 void trx_sys_close(void) {
-  ut_ad(srv_shutdown_state.load() == SRV_SHUTDOWN_EXIT_THREADS);
```

Comment:
same as above, why is it removed? can we fix the assert instead? what is the state on the error condition?

Reply:
@avbelov23 did you print srv_shutdown_state and check?